### PR TITLE
New enroll user method into canvas section

### PR DESCRIPTION
**Pull request by renaedahiya**
Hello. This pull request added the ability to enrollUsers into either a section or a course.

### CODE REVIEW

**Comment by kawesome** [comment]
Can one of the admins verify this patch?

File: src/main/java/edu/ksu/canvas/requestOptions/UnEnrollOptions.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package edu.ksu.canvas.requestOptions;
```
Comment (ToeBee) [inline src/main/java/edu/ksu/canvas/requestOptions/UnEnrollOptions.java]: Hmm this is a slight oddball. The other classes in this package wrap functionality to add multiple optional parameters to a request. This is just a straight up enum. But I'm not sure if there is a better place to put it either. So while it seems a little odd, I don't really have a suggested change here.

File: src/main/java/edu/ksu/canvas/impl/EnrollmentImpl.java [hunk 2]
```diff
@@ -52,7 +53,26 @@ public EnrollmentImpl(String canvasBaseUrl, Integer apiVersion, OauthToken oauth
 
     @Override
     public Optional<Enrollment> enrollUser(Enrollment enrollment) throws InvalidOauthTokenException, IOException {
-        String createdUrl = buildCanvasUrl("courses/" + enrollment.getCourseId() + "/enrollments", Collections.emptyMap());
+        return enrollUser(enrollment, false);
+    }
+
+    /**
+     * Enrolls a user into either a course or section based on the boolean isSectionEnrollment value.
+     *
```
[the remark this answers is not in the document]
Comment (renaedahiya) [reply src/main/java/edu/ksu/canvas/impl/EnrollmentImpl.java]: cool thanks

File: src/main/java/edu/ksu/canvas/impl/EnrollmentImpl.java [hunk 3]
```diff
@@ -52,7 +53,26 @@ public EnrollmentImpl(String canvasBaseUrl, Integer apiVersion, OauthToken oauth
...
+     * @param enrollment partially populated Enrollment object
+     * @param isSectionEnrollment when true, invoke canvas url to enroll a user in a section, otherwise enroll a user in a course.
+     * @return Optional<Enrollment>
+     * @throws InvalidOauthTokenException
+     * @throws IOException
+     */
+    @Override
+    public Optional<Enrollment> enrollUser(Enrollment enrollment, boolean isSectionEnrollment) throws InvalidOauthTokenException, IOException {
```
[the remark this answers is not in the document]
Comment (renaedahiya) [reply src/main/java/edu/ksu/canvas/impl/EnrollmentImpl.java]: Yes that sounds good to me.

File: src/main/java/edu/ksu/canvas/impl/EnrollmentImpl.java [hunk 4]
```diff
@@ -64,8 +84,13 @@ public EnrollmentImpl(String canvasBaseUrl, Integer apiVersion, OauthToken oauth
 
     @Override
     public Optional<Enrollment> dropUser(String courseId, String enrollmentId) throws InvalidOauthTokenException, IOException {
+        return dropUser(courseId, enrollmentId, UnEnrollOptions.DELETE.toString());
+    }
+
+    @Override
+    public Optional<Enrollment> dropUser(String courseId, String enrollmentId, String unEnrollOption) throws InvalidOauthTokenException, IOException {
```
[the remark this answers is not in the document]
Comment (renaedahiya) [reply src/main/java/edu/ksu/canvas/impl/EnrollmentImpl.java]: yes thanks

File: src/main/java/edu/ksu/canvas/requestOptions/UnEnrollOptions.java (hunk 1 above)
Comment (renaedahiya) [reply src/main/java/edu/ksu/canvas/requestOptions/UnEnrollOptions.java]: Yes, I noticed that too. Guess we can start with this and refactor down the road? Open to suggestions.

File: src/main/java/edu/ksu/canvas/impl/EnrollmentImpl.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (renaedahiya) [reply src/main/java/edu/ksu/canvas/impl/EnrollmentImpl.java]: I'm assuming we want to keep enrollUser(Enrollment enrollment) for backward compatibility too, does that sound good?

[the remark this answers is not in the document]
Comment (renaedahiya) [reply src/main/java/edu/ksu/canvas/impl/EnrollmentImpl.java]: Can mark that method deprecated.

[the remark this answers is not in the document]
Comment (ToeBee) [reply src/main/java/edu/ksu/canvas/impl/EnrollmentImpl.java]: I kind of want to say just get rid of it... but yeah marking as deprecated is probably a better call. I really don't know how many other people are using this code but trying not to break compilation frequently is a good idea.

**Comment by ToeBee** [comment]
Jenkins: ok to test

**Comment by ToeBee** [comment]
Ah, Jenkins pointed out a couple of other problems. Java 8 has gotten very picky with some javadoc things. 
* All of the `@throws` declarations need some text after them explaining when the method throws an exception
* the Oauth and illegal argument exceptions are runtime exceptions so they don't need to be explicitly declared
* It doesn't seem to like the `<Enrollment>` parts of `@return` statements. 

You can make maven check the javadoc stuff by running `mvn javadoc:javadoc`

File: src/main/java/edu/ksu/canvas/requestOptions/UnEnrollOptions.java (hunk 1 above)
Comment (renaedahiya) [reply src/main/java/edu/ksu/canvas/requestOptions/UnEnrollOptions.java]: Thanks ToeBee, looking now

Comment (renaedahiya) [reply src/main/java/edu/ksu/canvas/requestOptions/UnEnrollOptions.java]: Pushed the javadoc fixes, let me know how it goes!

**Comment by renaedahiya** [comment]
Morning ToeBee.  Is this branch ready to be merged?

**Comment by ToeBee** [comment]
Yeah, I think I will go ahead and merge this. Thanks for the pull request! Do you need this functionality to be released in a public release soon?

**Comment by renaedahiya** [comment]
Thank you for the merge. A public release on or before 10/05/17 would be greatly appreciated.

**Comment by ToeBee** [comment]
I just released 1.0.14

**Pull request merged**